### PR TITLE
REGISTRY-3114 Fix - Adding validation when creating an asset with an existing name and version

### PR DESCRIPTION
**Pull request by Rajith90**

### CODE REVIEW

File: modules/es-extensions/publisher/asset/schema/themes/default/js/create_form.js [hunk 1]
```diff
@@ -51,10 +51,48 @@ $(function() {
...
+            var version;
+            var name;
+            if (selectedValue == "upload") {
+                name = $('#schema_file_name').val();
+                version = $('#file_version').val();
+            } else {
+                name = $('input[name="overview_name"]').val();
+                version = $('input[name="overview_version"]').val();
+            }
+            var ajaxURL = caramel.context + '/apis/assets?type=schema&q="name":"' + name + '","version":"' + version + '"';
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/schema/themes/default/js/create_form.js]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/swagger/apis/swaggers.jag [hunk 2]
```diff
@@ -67,6 +67,19 @@ require('/modules/publisher.js').exec(function (ctx) {
...
+
+        var rxt = require('rxt');
+        var am = rxt.asset.createUserAssetManager(ctx.session, "swagger");
+        var query = {};
+        query.overview_name = swaggerFileName;
+
+        var assets = am.search(query);
+        for (var i = 0; i < assets.length; i++) {
+            if (assets[i].version == version) {
+                var msg = "Resource already exist with name \"" + swaggerFileName + "\" and version \"" + version + "\"";
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/swagger/apis/swaggers.jag]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/swagger/asset.js [hunk 3]
```diff
@@ -125,6 +125,21 @@ asset.manager = function(ctx) {
...
             properties[0][1] = version;
 
+            var rxt = require('rxt');
+            var am = rxt.asset.createUserAssetManager(ctx.session, this.type);
+            var query = {};
+            query.overview_name = name;
+            var assets = am.search(query);
+            for (var i = 0; i < assets.length; i++) {
+                if (assets[i].version == version) {
+                    var msg = "resource already exist with Name \"" + name + "\" and version \"" + version + "\"";
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/swagger/asset.js]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/swagger/themes/default/js/create_form.js [hunk 4]
```diff
@@ -49,10 +49,48 @@ $(function() {
...
+            var version;
+            var name;
+            if (selectedValue == "upload") {
+                name = $('#swagger_file_name').val();
+                version = $('#file_version').val();
+            } else {
+                name = $('input[name="overview_name"]').val();
+                version = $('input[name="overview_version"]').val();
+            }
+            var ajaxURL = caramel.context + '/apis/assets?type=swagger&q="name":"' + name + '","version":"' + version + '"';
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/swagger/themes/default/js/create_form.js]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/wadl/apis/wadls.jag [hunk 5]
```diff
@@ -67,6 +67,19 @@ require('/modules/publisher.js').exec(function (ctx) {
...
+
+        var rxt = require('rxt');
+        var am = rxt.asset.createUserAssetManager(ctx.session, "wadl");
+        var query = {};
+        query.overview_name = wadlFileName;
+
+        var assets = am.search(query);
+        for (var i = 0; i < assets.length; i++) {
+            if (assets[i].version == version) {
+                var msg = "Resource already exist with name \"" + wadlFileName + "\" and version \"" + version + "\"";
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/wadl/apis/wadls.jag]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/wadl/asset.js [hunk 6]
```diff
@@ -147,6 +147,21 @@ asset.manager = function(ctx) {
...
             properties[0][1] = version;
 
+            var rxt = require('rxt');
+            var am = rxt.asset.createUserAssetManager(ctx.session, this.type);
+            var query = {};
+            query.overview_name = name;
+            var assets = am.search(query);
+            for (var i = 0; i < assets.length; i++) {
+                if (assets[i].version == version) {
+                    var msg = "resource already exist with Name \"" + name + "\" and version \"" + version + "\"";
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/wadl/asset.js]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/wadl/themes/default/js/create_form.js [hunk 7]
```diff
@@ -51,10 +51,48 @@ $(function() {
...
+            var version;
+            var name;
+            if (selectedValue == "upload") {
+                name = $('#wadl_file_name').val();
+                version = $('#file_version').val();
+            } else {
+                name = $('input[name="overview_name"]').val();
+                version = $('input[name="overview_version"]').val();
+            }
+            var ajaxURL = caramel.context + '/apis/assets?type=wadl&q="name":"' + name + '","version":"' + version + '"';
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/wadl/themes/default/js/create_form.js]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/wsdl/apis/wsdls.jag [hunk 8]
```diff
@@ -67,6 +67,19 @@ require('/modules/publisher.js').exec(function (ctx) {
...
+
+        var rxt = require('rxt');
+        var am = rxt.asset.createUserAssetManager(ctx.session, "wsdl");
+        var query = {};
+        query.overview_name = wsdlFileName;
+
+        var assets = am.search(query);
+        for (var i = 0; i < assets.length; i++) {
+            if (assets[i].version == version) {
+                var msg = "Resource already exist with name \"" + wsdlFileName + "\" and version \"" + version + "\"";
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/wsdl/apis/wsdls.jag]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/wsdl/asset.js [hunk 9]
```diff
@@ -122,6 +122,21 @@ asset.manager = function(ctx) {
...
             properties[0][1] = version;
 
+            var rxt = require('rxt');
+            var am = rxt.asset.createUserAssetManager(ctx.session, this.type);
+            var query = {};
+            query.overview_name = name;
+            var assets = am.search(query);
+            for (var i = 0; i < assets.length; i++) {
+                if (assets[i].version == version) {
+                    var msg = "resource already exist with Name \"" + name + "\" and version \"" + version + "\"";
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/wsdl/asset.js]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/wsdl/themes/default/js/create_form.js [hunk 10]
```diff
@@ -49,10 +49,48 @@ $(function() {
...
+            var version;
+            var name;
+            if (selectedValue == "upload") {
+                name = $('#wsdl_file_name').val();
+                version = $('#file_version').val();
+            } else {
+                name = $('input[name="overview_name"]').val();
+                version = $('input[name="overview_version"]').val();
+            }
+            var ajaxURL = caramel.context + '/apis/assets?type=wsdl&q="name":"' + name + '","version":"' + version + '"';
```
Comment (sameerak) [inline modules/es-extensions/publisher/asset/wsdl/themes/default/js/create_form.js]: code line is longer than 120 characters.
Please break this in to 2 lines.

File: modules/es-extensions/publisher/asset/wadl/asset.js (hunk 6 above)
Comment (Rajith90) [inline modules/es-extensions/publisher/asset/wadl/asset.js]: less than 120 characters.

File: modules/es-extensions/publisher/asset/wsdl/apis/wsdls.jag (hunk 8 above)
Comment (Rajith90) [inline modules/es-extensions/publisher/asset/wsdl/apis/wsdls.jag]: less than 120 characters.

File: modules/es-extensions/publisher/asset/wsdl/asset.js (hunk 9 above)
Comment (Rajith90) [inline modules/es-extensions/publisher/asset/wsdl/asset.js]: less than 120 characters.

File: modules/es-extensions/publisher/asset/swagger/asset.js (hunk 3 above)
Comment (Rajith90) [inline modules/es-extensions/publisher/asset/swagger/asset.js]: less than 120 characters.

File: modules/es-extensions/publisher/asset/wsdl/themes/default/js/create_form.js (hunk 10 above)
Comment (Rajith90) [inline modules/es-extensions/publisher/asset/wsdl/themes/default/js/create_form.js]: less than 120 characters.

**Pull request merged**